### DISCUSSION TITLE
Horovod Trace shows a huge time gap between MPI_ALLTOALL and ALLTOALL

### DISCUSSION THREAD

**Discussion by kelvinzou** [horovod/horovod#2796] [Q&A]
**Environment:**
1. Framework: Tensorflow 
2. Framework version: 2.4.0
3. Horovod version: 0.20
4. MPI version: UCX+HPCX
7. Python version: 3.8
8. OS and version: Debian 9
9. GCC version:  6.3

I am using Horovod for my distributed CPU training, and I observed this issue:
There are a lot of MPI Alltoalls that take a very long time, but indeed they dont need to, they are stuck with ALLTOALL, but the real MPI_ALLTOALL is within a few milliseconds.

Not sure what can cause this?
Thank you

![image](https://user-images.githubusercontent.com/5438895/112427960-b5cf3b00-8cf7-11eb-81c1-499b6826dae4.png)
![image](https://user-images.githubusercontent.com/5438895/112428001-c2ec2a00-8cf7-11eb-976f-cedbbf1003ee.png)

**Comment by tgaddair** [comment]
Hey @kelvinzou, it looks like a lot of the time is being taken by negotiation (waiting for all the ranks to get to the same point). Is it possible that your workload is not evenly distributed among the workers?
